### PR TITLE
Create new field type: StringSearch

### LINKED ISSUE #29: String search type
Generic type to simulate an autocomplete on chat.

- bot asks a question
- user answers with some words
- bot calls search API to get the answer
  - if returns one result, ask for confirmation 
  - if returns multiple, ask for choose options using MultipleChoice
- user continues the flow with the new value from API

### PR DESCRIPTION
**Pull request by rafaelverger**
- [x] New multi-step field type to query external server with users input and show its results
- [x] Custom config for defined messages
- [x] Custom config for server's response translation
- [x] Improve readme with this new field
- [x] Tests

Closes #29

### CODE REVIEW

File: src/core/types.ts [hunk 1]
```diff
@@ -82,14 +92,99 @@ const types: { [name: string]: RuleType } = {
...
+      },
+      {
+        transform: async (results: any, rule: Rule, bot: YveBot) => {
+          const { messages } = rule.config;
+          let options, message;
+          if (results.length === 1) {
+            message = `${messages.didYouMean}: ${results[0].label}?`;
+            options = [
+              { label: 'Sim', value: results[0].value },
+              { label: 'Não', value: null },
```
Comment (andersonba) [inline src/core/types.ts]: Use translations here

**Comment by andersonba** [comment]
You forgot to add the `yes/no` messages in docs.
I was thinking so, we should have a config default with generic messages, like as "no found results", "type the word to search"...
What do you think about it?

**Comment by rafaelverger** [comment]
It makes sense, @andersonba and would those messages be able to be overridden by rule's configs?

**Comment by andersonba** [comment]
Yeah! Should be overriden.

**Pull request merged**